### PR TITLE
wiimms-iso-tools: 3.02a -> 3.05a

### LINKED ISSUE #213181: wiimms-iso-tools can't run some subcommands
### Describe the bug

This tool have too many commands / subcommands to check them all for failures. I ran into one that I had an issue with... `wit extract`. I haven't tried any other commands / subcommands and may not need to.

- `wit help` printed help just fine.
- `wit extract` exited with an error.

### Steps To Reproduce
1. Obtain a GC/Wii image.
2. Use commands like `wit extract Game.iso Game-Dir`.
3. Get an error instead of Game folder.

### Expected behavior

Run `wit extract` commands to extract GC/Wii images without issues.

### Screenshots

AN output of `wit extract` when I tried to extract literally anything. Nothing worked.
```
❯ wit extract Game.iso Game-Dir
*****  wit: Wiimms ISO Tool v3.02a r0 x86_64 - Dirk Clemens - 2023-01-05  *****
!! wit: ERROR #126 [FATAL ERROR] in wd_read_part() @ src/libwbfs/wiidisc.c#865
!!      ASSERTION FAILED !!!
```

### Additional context

I tried Archlinux next (`v3.05a`). No issue.
I think it is a source code issue with getting this running on Nix and/or NixOS.

### Notify maintainers

@nilp0inter 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.7, NixOS, 23.05 (Stoat), 23.05pre447289.9b97ad7b433`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `"nixos"`
 - channels(chris): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by winterqt**
###### Description of changes

https://wit.wiimm.de/changelog.html#r8245
https://wit.wiimm.de/changelog.html#r8427
https://wit.wiimm.de/changelog.html#r8638

This also adds Clang (and, by extension, macOS) support.

May fix #213181.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lasers** [comment]
This fixes https://github.com/NixOS/nixpkgs/issues/213181. Thank you! @winterqt

**Comment by winterqt** [comment]
cc @nilp0inter

**Comment by nilp0inter** [comment]
This looks great. Thank you @winterqt and @lasers.

**Comment by winterqt** [comment]
I re-added the ncurses patch (I thought it was merged), as well as removed the CFLAGS (`-Wno-error=format-security` can be dropped because of the ncurses patch (not sure why it wasn't dropped when the patch was added), and upstream has fixed `-fno-common` I assume, because it works without it.)

PTAL, @nilp0inter.

File: pkgs/tools/filesystems/wiimms-iso-tools/default.nix [hunk 1]
```diff
@@ -28,10 +28,10 @@ stdenv.mkDerivation rec {
     patchShebangs setup.sh
     patchShebangs gen-template.sh
     patchShebangs gen-text-file.sh
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/wiimms-iso-tools/default.nix:30]:
```suggestion
    patchShebangs setup.sh gen-template.sh gen-text-file.sh
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by winterqt** [comment]
@NickCao I'd appreciate it if you just poked me again about this before squashing and merging like that... that commit message is pretty poorly formed due to GitHub's automatic filling. Thanks.

**Comment by NickCao** [comment]
I would take notice next time, hope github just give me an `amend` button.

**Comment by winterqt** [comment]
I'd really prefer if you just reminded me instead of doing that... it's not like I'm inactive. This is my PR, after all >.<

**Pull request merged**